src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -463,7 +463,7 @@ private static unsafe bool TryParseNumber(ref char* str, char* strEnd, NumberSty
                         {
                             number.Scale = 0;
                         }
-                        if ((state & StateDecimal) == 0)
+                        if ((number.Kind != NumberBufferKind.FloatingPoint) && (state & StateDecimal) == 0)
```

Comment:
`System.Decimal` treats `-0` as an integer negative zero. I could get rid of this if we think it is fine to change as well.

Reply:
Sorry, thought I linked it. It is in the UTF8Parser: https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Parser/Utf8Parser.Decimal.cs#L67

Reply:
> I could get rid of this if we think it is fine to change as well.

I think it is fine to change. It is inline with the other corner case handling we are fixing.

Reply:
Fixed.